src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+    @check_normalized
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super(BasisSet).normalize(archive, logger)
+        self.n_orbitals = len(self.orbitals)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_normalized = False
+
+
+class MuffinTinRegion(BasisSet, Mesh):
+    """
+    Muffin-tin region around atoms, containing the augmented part of the APW basis set.
+    The latter is structured by l-channel. Each channel contains a base (S)(L)APW definition,
+    which may be extended via local orbitals.
+    """
+
+    # there are 2 main ways of structuring the APW basis set
+    # either as APW and lo in the MT region
+    # or by l-channel in the MT region
+
+    radius = Quantity(
+        type=np.float64,
+        shape=[],
+        unit='meter',
+        description="""
+        The radius descriptor of the `MuffinTin` is spherical shape.
+        """,
+    )
+
+    l_max = Quantity(
+        type=np.int32,
+        description="""
+        Maximum angular momentum quantum number that is sampled.
+        Starts at 0.
+        """,
+    )
+
+    l_channels = SubSection(sub_section=APWLChannel.m_def, repeats=True)
+
+    def _determine_apw(self) -> dict[str, int]:
+        """
+        Aggregate the APW component count in the muffin-tin region.
+        Invokes `normalize` on `l_channels`.
+        """
+        for l_channel in self.l_channels:
+            l_channel.normalize(None, None)
+
+        type_count: dict[str, int]
+        if len(self.l_channels) > 0:
+            l_channel = self.l_channels[0]
+            # dynamically determine `type_count` structure
+            type_count = l_channel._determine_apw(l_channel.orbitals)
+            for channel in self.l_channels[1:]:
+                type_count.update(channel._determine_apw())
+        return type_count
+
+    @check_normalized
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class BasisSetContainer(NumericalSettings):
+    """
+    A section defining the full basis set used for representing the electronic structure
+    during the diagonalization of a Hamiltonian (component), as defined in `ModelMethod`.
+    This section may contain multiple basis set specifications under the basis_set_components,
+    each with their own parameters.
+    """
+
+    native_tier = Quantity(
+        type=str,  # to be overwritten by a parser `MEnum`
+        description="""
+        Code-specific tag indicating the overall precision based on the basis set parameters.
+        The naming conventions of the same code are used. See the parser implementation for the possible values.
+        The number of tiers varies, but a typical example would be `low`, `medium`, `high`.
+        """,
+    )
+
+    # TODO: add reference to `electronic_structure`,
```

Comment:
Delete this block comment

Reply:
This is something that will be for another PR, given that we don't have any place yet for the actual eletronic object.
It looks like with the C4 task (@EBB2675), this may be incoporated. It is even possible that the referencing direction is the inverse of how I stipulate it here.

Reply:
What is an "electronic object"?